src/lib/cache.ts

```diff
@@ -0,0 +1,67 @@
+import * as fs from "fs";
+import * as path from "path";
+import { hash } from "./hash";
+
+export class Cache {
+  private inMemoryCache: Record<string, string> = {};
+  private diskCacheFolderPath: string = "./.cache";
+  private diskCacheFileName: string = "cache.json";
```

Comment:
An optimisation if the cache gets big is to store different requests in different cache files